### ISSUE TITLE
MauiMaps Android removed pin still appears on map

### ISSUE THREAD

**Issue by davefxy** [VladislavAntonyuk/MauiSamples#250]
When I attempt to remove pin on MauiMaps Android, removed pin still appears on map.
I noticed in Android CustomMapHandler Markers is never is set to anything.
I added the following lines to AddPins() after the check for null at start of methos:
`        if (Markers == null)
            Markers = new List<Marker>();
`
Then the pin gets removed from map as expected.

Another issue I see and dont understand is the Action finished method for ImageSource.LoadImage appears to be called 4 times for each ImageSource.LoadImage () request. I examined Markers after adding a pin and there are 4 new Marker entries for an added pin.
I am using the latest version on the repository and Microsoft Visual Studio Community 2022 (64-bit) - Preview
Version 17.6.0 Preview 2.0

**Comment by VladislavAntonyuk** [comment]
AddPins is only called in MapPins. in MapPins I clear all markers: 

https://github.com/VladislavAntonyuk/MauiSamples/blob/0efbfec1590e630b626ddbe26d85832dc617b6e5/MauiMaps/Platforms/Android/CustomMapHandler.cs#L40

LoadImage is called and the app continues running. Then in the background image is loading and when ready the Action callback is executed. The marker may appear in a few seconds after you added Pin, depending on the image source (e.g. load image from the internet with low speed)

**Comment by davefxy** [comment]
I think I determined what was happening and made the following changes to CustomMapHandler. Just rely on the Google map. The Markers List is not needed.
In MapPins() : remove mapHandler.Markers.Clear();
In AddPins() : add Map.Clear(); before foreach (var pin in mapPins)
In AddMarker(: remove markers?.Add(marker);

**Comment by davefxy** [comment]
Using the Map.Clear() also prevents the GoogleMap markers from accumulating when adding/deleting pins since these are never cleared otherwise.

**Comment by VladislavAntonyuk** [comment]
Why do you say they never cleaned if they actually do? On sample app click Add and Remove multiple times to see that pins disappear.
Once again markers collection represents pins collection. If you remove pin you remove marker. If you remove marker it will be remove from map.

**Comment by davefxy** [comment]
Maybe we are running different versions of Maui. I use Microsoft Visual Studio Community 2022 (64-bit) - Preview Version 17.6.0 Preview 2.0.
I created a new app.: MauiMaps. I copied your code to this new app. I added the same nugets as your sample.
I run this app. I Added 2 new pins. I now have 4 pins displayed on map.  I Removed 2 pins. There are still 4 pins displayed. Whenever a pin is added, it is added to Markers and Map.AddMarker. When I remove a pin, mapHandler.Markers.Clear(); is called. This clears the local Marker collection, and then adds the map.pins. However what clears the GoogleMap collection?
I see the same result when I compile and run your MauiMaps sample as is.
